source/fundamentals/bson.txt

```diff
@@ -0,0 +1,301 @@
+         	Street string
+         	City   string
+         	State  string
+         }
+                  
+         type Student struct {
+         	FirstName string  `bson:"first_name,omitempty"`
+         	LastName  string  `bson:"last_name,omitempty"`
+         	Address   Address `bson:"inline"`
+         	Age       int
+         }
+         ...
+
+         coll := client.Database("school").Collection("students")
+         address1 := Address{ "1 Lakewood Way", "Elwood City", "PA" }
+         student1 := Student{ FirstName : "Arthur", Address : address1, Age : 8}
+         _, err = coll.InsertOne(context.TODO(), student1)
+
+      The corresponding BSON representation looks like this:
+
+      .. code-block:: json
+         :copyable: false
+      
+         { 
+           "_id" : ObjectId("..."),
+           "first_name" : "Arthur",
+           "street" : "1 Lakewood Way",
+           "city" : "Elwood City",
+           "state" : "PA",
+           "age" : 8
+         }
+
+      In this example, struct tags make the driver:
+      
+      - Set custom BSON field names such as ``first_name``
+      - Omit the empty ``LastName`` field
+      - Flatten the nested struct and bring all fields up to the top
+        level
+
+      .. tabs::
+
+   .. tab:: No Struct Tags
+      :tabid: no-struct-tags
+
+      The following example demonstrates how the Go Driver marshals a
+      struct without any struct tags:
+
+      .. code-block:: go
+
+         type Address struct {
+         	Street string
+         	City   string
+         	State  string
+         }
+                  
+         type Student struct {
+         	FirstName string
+         	LastName  string
+         	Address   Address
+         	Age       int
+         }
+         ...
+
+         coll := client.Database("school").Collection("students")
+         address1 := Address{ "1 Lakewood Way", "Elwood City", "PA" }
+         student1 := Student{ FirstName : "Arthur", Address : address1, Age : 8}
+         _, err = coll.InsertOne(context.TODO(), student1)
+
+      The corresponding BSON representation looks like this:
+
+      .. code-block:: json
+         :copyable: false
+      
+         { 
+           "_id" : ObjectId("..."),
+           "firstname" : "Arthur",
+           "lastname" : "",
+           "address": {
+                        "street" : "1 Fern Way",
+                        "city" : "Elwood City",
+                        "state" : "PA"
+                      },
+           "age" : 8
+         }
+         
+      Without struct tags, the driver:
+      
+      - Sets the lowercase of the struct fields as the BSON field names
+      - Includes an empty ``lastname`` field
+      - Stores the ``Address`` field as a nested object
+
+Unmarshalling
+-------------
+
+You can unmarshal BSON documents using the ``Decode()`` method. When you
+execute either a ``FindOne()`` or ``Find()`` query, you can use the
+``Decode()`` method on the return object to unmarshal the results.
```

Comment:
This sentence is a bit confusing. You can only call `Decode()` directly on the result of `FindOne()`; `Find()` returns `(*mongo.Cursor, error)`, so you can't call `Decode()` directly on the result. Also, you can call `Decode()` on any cursor, not just cursors returned by `Find()` (e.g. `Aggregate()` also returns a cursor).

Consider updating the description to say you can call `Decode()` on the result of `FindOne()` or any `*mongo.Cursor`.